recipe/2024-01-global-scope-secret-store.md

```diff
@@ -0,0 +1,259 @@
+# Adding support to extend the capabilities of secretStores to a global scope.
+
+* **Status**: Pending
+* **Author**: Vishwanath Hiremath (@vishwahiremat)
+
+## Overview
+
+Today we have Application.Core/secretStores resource to securely manage secrets for the Application. However a gap exists if user wants to create a secret before application or environment is created. To address this limitation, we need to enhance the capabilities of the existing secretStores resource. The objective is to extend its support to a global scope, enabling users to store secrets prior to the creation of the application or environment.
+
+
+## Terms and definitions
+| Term     | Definition                                                                                                                                                                                                 |
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Global scope | The ability of the secretStores resource to operate universally, allowing users to manage secrets independently of a particular application or environment. |
+| secretStores | Resource used to securely manage secrets for Environment and Application. |
+| Private Terraform Repository | A private Terraform repository typically refers to a version control repository that contains Terraform module code, but is not publicly accessible. | 
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/7030
+
+### Goals
+
+- Enable support to extend the capabilities of secretStores to a global scope.
+
```

Comment:
Something I am curious about is whether this type should move to UCP. `Applications.Core` defines it as a part of the Radius RP rather than part of UCP.

Reply:
If we have usecases for this type in other parts of UCP then it would be better to rename/move than it would be to have two different secret types.